models/cart/src/cart-draft/generator.ts

```diff
@@ -0,0 +1,50 @@
+import { Reference, AddressDraft } from '@commercetools-test-data/commons';
+import { fake, Generator, oneOf } from '@commercetools-test-data/core';
+import * as LineItem from '@commercetools-test-data/line-item';
+import {
+  inventoryMode,
+  origin,
+  shippingMode,
+  taxCalculationMode,
+  taxMode,
+  taxRoundingMode,
+} from '../constants';
+import { TCartDraft } from '../types';
+
+// https://docs.commercetools.com/api/projects/carts#cartdraft
+
+const generator = Generator<TCartDraft>({
+  fields: {
+    currency: oneOf('EUR', 'USD'),
+    key: fake((f) => f.lorem.slug(2)),
+    customerId: fake((f) => f.datatype.uuid()),
+    customerEmail: fake((f) => f.internet.email()),
+    customerGroup: fake(() => Reference.random().typeId('customer-group')),
+    anonymousId: fake((f) => f.datatype.uuid()),
+    businessUnit: fake(() => Reference.random().typeId('business-unit')),
+    store: null,
+    country: fake((f) => f.address.countryCode()),
+    inventoryMode: oneOf(...Object.values(inventoryMode)),
+    taxMode: oneOf(...Object.values(taxMode)),
+    taxRoundingMode: oneOf(...Object.values(taxRoundingMode)),
+    taxCalculationMode: oneOf(...Object.values(taxCalculationMode)),
+    lineItems: fake(() => [LineItem.LineItemDraft.random()]),
+    customLineItems: [],
+    shippingAddress: fake(() => AddressDraft.random()),
+    billingAddress: fake(() => AddressDraft.random()),
+    shippingMethod: fake(() => Reference.random().typeId('shipping-method')),
+    externalTaxRateForShippingMethod: null,
+    custom: null,
+    locale: fake((f) => f.random.locale()),
+    deleteDaysAfterLastModification: null,
+    shippingRateInput: null,
+    origin: oneOf(...Object.values(origin)),
+    shippingMode: oneOf(...Object.values(shippingMode)),
+    customShipping: null,
+    shipping: [],
+    itemShippingAddresses: [],
+    discountCodes: fake((f) => [f.random.word()]),
```

Comment:
I think this'll generate the whole discount code rather than just its string `code`, but you raise a good point that we should just `.build()` off it and access the code for completeless!

like `[DiscountCode.random().build().code]` or something 🤔  Not sure how invoking `build` in this will impact the transformer but I'll give it a shot!

Reply:
Just spent some time here. Seems like doing this "the right way" might be more effort than it's worth 😬